app/javascript/src/Common/components/SelectWithModal.tsx

```diff
@@ -126,12 +120,18 @@ const SelectWithModal = <T extends IRecord>({
     setPage(1)
   }
 
+  const clearResults = () => {
+    setQuery('')
+    setSearchResults([], 0)
+  }
+
   const handleModalOnSetPage = (newPage: number) => {
     setPage(newPage)
   }
 
   const handleOnModalClose = () => {
     setModalOpen(false)
+    clearResults()
```

Comment:
Without this, when modal is closed, and then reopened, old results were showing, despite the search input already being empty.